### PR TITLE
Recent vessels modal

### PR DESCRIPTION
**Pull request by andresgnlez**
TODO:

- [ ] Implement localStorage

### CODE REVIEW

File: app/src/actions/vesselInfo.js [hunk 1]
```diff
@@ -55,6 +57,13 @@ export function setCurrentVessel(seriesGroup) {
           seriesgroup: data.seriesgroup
         }
       });
+
+      dispatch({
```
Comment (nerik) [inline app/src/actions/vesselInfo.js]: DRY
`dispatch(setRecentVesselHistory())`

File: app/src/actions/vesselInfo.js [hunk 2]
```diff
@@ -99,6 +108,13 @@ export function setPinnedVessels(pinnedVessels) {
           type: LOAD_PINNED_VESSEL,
           payload: Object.assign({}, pinnedVessel, data)
         });
+
+        dispatch({
```
Comment (nerik) [inline app/src/actions/vesselInfo.js]: DRY
`dispatch(setRecentVesselHistory())`

File: app/src/reducers/vesselInfo.js [hunk 3]
```diff
@@ -12,15 +12,21 @@ import {
...
 import { DEFAULT_TRACK_HUE } from 'constants';
 
 const initialState = {
   tracks: [],
   details: [],
   detailsStatus: null,
-  pinnedVesselEditMode: false
+  pinnedVesselEditMode: false,
+  history: [],
+  recentVesselModal: {
```
Comment (nerik) [inline app/src/reducers/vesselInfo.js]: I would move this to the map reducer. I don't like all modals being there, but we either keep them there, or  refactor *all* of them into a new reducer.

File: app/src/reducers/vesselInfo.js [hunk 4]
```diff
@@ -171,6 +178,46 @@ export default function (state = initialState, action) {
         details: [...state.details.slice(0, detailsIndex), newDetails, ...state.details.slice(detailsIndex + 1)]
       });
     }
+
+    case SET_RECENT_VESSELS_VISIBILITY: {
+      const newState = Object.assign({}, state);
+      newState.recentVesselModal = {
```
Comment (nerik) [inline app/src/reducers/vesselInfo.js]: I would move this to the map reducer (see above)

**Comment by nerik** [comment]
@andresgnlez Please do the localStorage part in another PR

**Comment by andresgnlez** [comment]
@nerik Agree about modals. Actually, there is already a task about it: https://www.pivotaltracker.com/story/show/138139479

**Pull request merged**